tests/integration/GameStateMachine/Skeleton.tla

```diff
@@ -7,6 +7,18 @@ vars ==
 PlutusInteger == 0..6
 PlutusByteString == {pk1, pk2, pk3, pk4, pk5}
 SHA2(x) == x
+
+RECURSIVE PlutusDataBounded(_)
+PlutusDataBounded(n) == 
+  IF n = 0
+  THEN {}
+  ELSE [ cons : {"Constr"}, arg0 : PlutusInteger, arg1 : PlutusDataBounded(n-1) ]
+
+PlutusData == UNION { PlutusDataBounded(n) : n \in 0 .. MAXDEPTH }
+
+ConstrData(n,i) == [cons |-> "Constr", arg0 |-> n, arg1 |-> i]
+MkNilData(x) == [cons |-> "Nil" ]
+
```

Comment:
It seems that `PlutusData` is a list of `Int`, but the `Nil` case is missing in the `PlutusDataBounded` function.

Reply:
`PlutusData` is supposed to be a value of type `Data`, which is being reified as a `DefaultFun` [here](https://github.com/input-output-hk/plutus/blob/014fc0e7ca4aa8c830c651f7f1f960829ed3dfa9/plutus-core/plutus-core/src/PlutusCore/Default/Builtins.hs#L92), the semantics of each of the constructors are [here](https://github.com/input-output-hk/plutus/blob/014fc0e7ca4aa8c830c651f7f1f960829ed3dfa9/plutus-core/plutus-core/src/PlutusCore/Default/Builtins.hs#L357)

I decided to keep only the constructors that are needed to make the contract work for two reasons: (A) they might change the plutus `DefaultFun` again and (B) we will need to do it anyway when we write `Plutus.tla`